### PR TITLE
[frontend] Shop Route fix

### PR DESCRIPTION
**Pull request by UGisBusy**
There's a issue that the function associate to "user view shop" before is not complete and wrong.

### Changes
- user can now view product list in home page
- user can navigate to shop where the product is from
- the shop page now show the real things ( in ur db )
  - shop icon, shop desc, products count
  - click view shop to view product ( and click the product item to view detail
  - click coupon to view shop coupon

### Testing Approach (if unconventional)
```sh
docker compose up --build
```
- click ``more`` in the product in main page
- go to the shop by clicking ``view shop`` from the detail product
- check product and shop coupon in the shop
- maybe edit seller's things to see if it shown on the shop page. path is ``shop/{sellerName}``.

### Additional context
~~ill add ``add to cart`` function after this merge.~~
done, just in my stash stack.

### CODE REVIEW

File: frontend/src/components/CouponItemShowShop.tsx [hunk 1]
```diff
@@ -18,57 +23,37 @@ interface ICouponItemShow {
...
   const handleClose = () => setShow(false);
 
-  const {
-    data: detailCouponData,
-    status,
-    refetch,
-  } = useQuery({
-    queryKey: ['sellerGetCouponDetail', data.id],
+  const { data, status, refetch } = useQuery({
+    queryKey: ['userGetSjopCouponDetail', couponId],
```
Comment (NOOBDY) [inline frontend/src/components/CouponItemShowShop.tsx]:
```suggestion
    queryKey: ['userGetShopCouponDetail', couponId],
```

**Review by NOOBDY — DISMISSED** [review]

File: frontend/src/pages/user/shop/index.tsx [hunk 2]
```diff
@@ -1,36 +1,97 @@
...
-                    <b>{userData.name}</b>
+                    <b>{shop.name}</b>
                   </h4>
                 </div>
                 <hr className='hr' />
-                <div className='center'> Products : {goodsData.length} items</div>
-                <TButton text='Explore Shop' action='/sellerID/shop' />
-                <TButton text='Check Coupons' action='/sellerID/coupons' />
+                <div className='center'> Products : {products.length} items</div>
+                <TButton text='Explore Shop' action={`/shop/${sellerName}/shop`} />
```
Comment (UGisBusy) [inline frontend/src/pages/user/shop/index.tsx]:
```suggestion
                <TButton text='Explore Shop' action={`/shop/${sellerName}/products`} />
```

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -97,15 +97,13 @@ function App() {
               <Route path=':coupon_id' element={<EachSellerCoupon />} />
             </Route>
 
-            <Route path='/:sellerName' element={<UserViewShop />}>
+            <Route path='shop/:sellerName' element={<UserViewShop />}>
               <Route index element={<Shop />} />
               <Route path='shop' element={<Shop />} />
```
Comment (UGisBusy) [inline frontend/src/App.tsx]:
```suggestion
              <Route path='products' element={<Shop />} />
```

**Review by UGisBusy — COMMENTED** [review]
metro mobile coding

**Comment by UGisBusy** [comment]
~~shit i fucked up~~
nvm all good

**Review by Error0229 — DISMISSED** [review]
womm

File: frontend/src/pages/discover/[goodsID].tsx [hunk 4]
```diff
@@ -8,10 +8,28 @@ import { useNavigate } from 'react-router-dom';
...
 import { CheckFetchStatus, RouteOnNotOK } from '@lib/Status';
 import { GoodsImgStyle } from '@pages/user/seller/allProducts/NewGoods';
 import { useAuth } from '@lib/Auth';
+import { formatDate } from '@lib/Functions';
+
+interface IProduct {
+  description: string;
+  expire_date: string;
+  id: number;
+  image_url: string;
```
Comment (KatoToyoki) [inline frontend/src/pages/discover/[goodsID].tsx]:
```suggestion
 product_image_url: string;
```

File: frontend/src/pages/discover/[goodsID].tsx [hunk 5]
```diff
@@ -43,52 +61,46 @@ const EachGoods = () => {
...
 
+  const expireDate = formatDate(data.expire_date);
   return (
     <div style={{ padding: '55px 12% 0 12%' }}>
       <Row>
         <Col xs={12} md={5} style={LeftBgStyle}>
           <div className='flex_wrapper' style={{ padding: '0 8% 10% 8%' }}>
             <div style={{ overflow: ' hidden' }}>
-              <img src={data.product_info.image_url} alt='File preview' style={GoodsImgStyle} />
+              <img src={data.image_url} alt='File preview' style={GoodsImgStyle} />
```
Comment (KatoToyoki) [inline frontend/src/pages/discover/[goodsID].tsx]:
```suggestion
              <img src={data.product_image_url} alt='File preview' style={GoodsImgStyle} />
```

**Review by KatoToyoki — CHANGES_REQUESTED** [review]

**Review by KatoToyoki — APPROVED** [review]

**Comment by UGisBusy** [comment]
> [UGisBusy](https://github.com/UGisBusy) requested a review from [KatoToyoki](https://github.com/KatoToyoki) [now](https://github.com/jykuo-love-shiritori/twp/pull/141#event-11420725729)

wrong person lul

**Review by KatoToyoki — DISMISSED** [review]

**Review by Error0229 — APPROVED** [review]

**Review by KatoToyoki — APPROVED** [review]

**Pull request merged**